g4me/analysis/PrepareTracksForMatchingAndFit_embedding.C

```diff
@@ -0,0 +1,330 @@
+    trackCandidatesHitCovMID.Clear();
+    particlesITS.Clear();
+    idTrackITS.clear();
+    idTrackMID.clear();
+
+    Int_t nTracks = io_underlaying.tracks.n + io_signal.tracks.n;
+    Int_t nHits = io_underlaying.hits.n + io_signal.hits.n;
+
+    std::vector<TClonesArray> allTracksHitPosITS(nTracks,TClonesArray("TVector3"));
+    std::vector<TClonesArray> allTracksHitCovITS(nTracks,TClonesArray("TMatrixDSym"));
+
+    std::vector<int> arrayHitID_MIDLayer1(nHits,-1);
+    std::vector<int> arrayHitID_MIDLayer2(nHits,-1);
+    nHits_MIDLayer1 = 0;
+    nHits_MIDLayer2 = 0;
+
+    //--------------------------------------------------------------------------
+    // Loop over pythia hits
+    for (int iHit=0; iHit<io_underlaying.hits.n; iHit++) {
+
+      //filling arrays of hit IDs from MID layers (coming from any charged tracks)
+
+      auto trackID = io_underlaying.hits.trkid[iHit];
+
+      if (!(IsTrackCharged(&io_underlaying, trackID))) continue;
+
+      mom.SetXYZ(io_underlaying.hits.px[iHit],io_underlaying.hits.py[iHit],io_underlaying.hits.pz[iHit]);
+      if (mom.Mag() < hitMinP) continue;
+
+      if (io_underlaying.hits.lyrid[iHit] == idLayerMID1) arrayHitID_MIDLayer1[nHits_MIDLayer1++] = iHit;
+      if (io_underlaying.hits.lyrid[iHit] == idLayerMID2) arrayHitID_MIDLayer2[nHits_MIDLayer2++] = iHit;
+
+      // filling arrays of hits from ITS tracks (only for interesting tracks: charged and primary).
+      // Hits from ITS are by definition all the hits having radius < rMaxITS
+      if (!(IsTrackInteresting(&io_underlaying, trackID))) continue;
+
+      pos.SetXYZ(gRandom->Gaus(io_underlaying.hits.x[iHit],resolutionITS),gRandom->Gaus(io_underlaying.hits.y[iHit],resolutionITS),gRandom->Gaus(io_underlaying.hits.z[iHit],resolutionITS));
+      if (pos.Perp() < rMaxITS) {
+	       new ((allTracksHitPosITS.at(trackID))[(allTracksHitPosITS.at(trackID)).GetEntries()]) TVector3(pos);
+	       new ((allTracksHitCovITS.at(trackID))[(allTracksHitCovITS.at(trackID)).GetEntries()]) TMatrixDSym(covITS);
+      }
+    }
+
+    // filling the final arrays with the hit information from good ITS tracks
+
+    nPreparedTracksITS = 0;
+
+    for (int iTrack=0; iTrack<io_underlaying.tracks.n; iTrack++) {
+      if (IsTrackInteresting(&io_underlaying, iTrack)) {
+	       new (trackCandidatesHitPosITS[nPreparedTracksITS]) TClonesArray(allTracksHitPosITS.at(iTrack));
+	       new (trackCandidatesHitCovITS[nPreparedTracksITS]) TClonesArray(allTracksHitCovITS.at(iTrack));
+	       idTrackITS.emplace_back(iTrack);
+	       TParticle part;
+	       part.SetPdgCode(io_underlaying.tracks.pdg[iTrack]);
+	       part.SetProductionVertex(io_underlaying.tracks.vx[iTrack],io_underlaying.tracks.vy[iTrack],io_underlaying.tracks.vz[iTrack],io_underlaying.tracks.vt[iTrack]);
+	       part.SetMomentum(io_underlaying.tracks.px[iTrack],io_underlaying.tracks.py[iTrack],io_underlaying.tracks.pz[iTrack],0.001*io_underlaying.tracks.e[iTrack]);
```

Comment:
I think the 0.001 factor for the particle energy should be removed (also from the PrepareTracksForMatchingAndFit.C macro, I'll take care of it)